x/stakingplus/keeper/msg_server_test.go

```diff
@@ -0,0 +1,73 @@
+package keeper_test
+
+import (
+	"github.com/cometbft/cometbft/crypto/secp256k1"
+	"github.com/golang/mock/gomock"
+
+	"cosmossdk.io/math"
+
+	simtestutil "github.com/cosmos/cosmos-sdk/testutil/sims"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+func (s *KeeperTestSuite) TestMsgCreateValidator() {
+	// approve any msg
+	s.stakingMsgServer.
+		EXPECT().
+		CreateValidator(gomock.Any(), gomock.Any()).
+		Return(&stakingtypes.MsgCreateValidatorResponse{}, nil).
+		AnyTimes()
+
+	testCases := map[string]struct {
+		valid     bool
+		acceptRet error
+	}{
+		"valid request": {
+			valid:     true,
+			acceptRet: nil,
+		},
+		"no grant found": {
+			valid:     false,
+			acceptRet: sdkerrors.ErrUnauthorized,
+		},
+	}
+
+	for name, tc := range testCases {
+		s.Run(name, func() {
+			s.foundationKeeper.
+				EXPECT().
+				Accept(gomock.Any(), gomock.Any(), gomock.Any()).
```

Comment:
You can make more sophisticated test if you use gomock.Eq(dummyGrantee), gomock.Eq(req).
req is the msg you generate below. you can use dummy msg for failed case. But it's trivial advice so you can merge.